thirdparty/DiskANN/src/distance.cpp

```diff
@@ -5,20 +5,30 @@ namespace diskann {
   template<typename T>
   DISTFUN<T> get_distance_function(diskann::Metric m) {
     if (m == diskann::Metric::L2) {
-      return [](const T* x, const T* y, size_t size) -> T {
+      return [](const T* x, const T* y, size_t size) -> float {
         float res = 0;
-        for (size_t i = 0; i < size; i++) {
-          res += ((float) x[i] - (float) y[i]) * ((float) x[i] - (float) y[i]);
+        auto  x_fp32 = std::make_unique<float[]>(size);
```

Comment:
So, every invokation of distance calculation will trigger two vector allocations? O_o
Should I add fp16 and bf16 distance calculations into Faiss instead?

Reply:
fp16/bf16 distance function will implemented later, like faiss::fvec_inner_product and fvec_L2sqr.